common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/RetryingBlockTransferor.java

```diff
@@ -274,7 +287,13 @@ private void handleBlockTransferFailure(String blockId, Throwable exception) {
       synchronized (RetryingBlockTransferor.this) {
         if (this == currentListener && outstandingBlocksIds.contains(blockId)) {
           if (shouldRetry(exception)) {
-            initiateRetry(exception);
+            try {
+              initiateRetry(exception);
+            } catch (Throwable t) {
+              logger.error("Exception while trying to initiate retry", t);
```

Comment:
Could you inline this common try-catch handler into `initiateRetry`?

Reply:
Thank you for the suggestion, refactored accordingly: [32498ff](https://github.com/apache/spark/pull/42426/commits/32498ff9a906a21746b619f1ddd316777d188417)